### PR TITLE
Shipkit Integration

### PR DESCRIPTION
**Pull request by kxu1026**
More doc problems are required to be fixed due to the shipkit integration. So, the PR is a little bit longer.
But, there is no logic change at all.

### CODE REVIEW

File: gradle/shipkit.gradle [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+           // Ensure that the release machine or Travis CI has this env variable exported
+           key = System.getenv("BINTRAY_API_KEY")
+
+           pkg {
+               repo = 'maven'
+               user = 'kxu'
+               userOrg = 'linkedin-coral'
+               name = 'coral'
+               licenses = ['BSD 2-Clause']
+               labels = ['coral', 'UDF', 'user defined functions', 'portable']
```
Comment (wmoustafa) [inline gradle/shipkit.gradle:35]: ['coral', 'sql', 'presto', 'spark', 'hive', 'views']

File: version.properties [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#Version of the produced binaries. This file is intended to be checked-in.
+#It will be automatically bumped by release automation.
+version=1.0.1
+previousVersion=1.0.0
```
Comment (wmoustafa) [inline version.properties:4]: Should the first version be 1.0.0?

File: coral-hive/build.gradle [hunk 3]
```diff
@@ -1,5 +1,5 @@
 apply plugin: 'java'
-apply plugin: 'maven-publish'
+//apply plugin: 'maven-publish'
```
[the remark this answers is not in the document]
Comment (kxu1026) [reply coral-hive/build.gradle:2]: woops. Missed this part. removing.

File: coral-hive/build.gradle [hunk 4]
```diff
@@ -36,36 +36,6 @@ dependencies {
...
-task sourcesJar(type: Jar) {
-  from sourceSets.main.output
-  archiveClassifier = 'sources'
-}
-
-task javadocJar(type: Jar) {
-  from javadoc
-  archiveClassifier = 'javadoc'
-}
-
```
[the remark this answers is not in the document]
Comment (kxu1026) [reply coral-hive/build.gradle:68]: After shipkit integration, All these sections are covered by shipkit. Without removing them, the compile will complain the duplication of tasks.

File: coral-hive/src/main/java/com/linkedin/coral/hive/hive2rel/HiveDbSchema.java [hunk 5]
```diff
@@ -83,7 +83,7 @@ public RelProtoDataType getType(String s) {
 
   /**
    * A Hive DB does not have subschema
-   * @param name
+   * @param name {@link String} of the sub schema
```
[the remark this answers is not in the document]
Comment (kxu1026) [reply coral-hive/src/main/java/com/linkedin/coral/hive/hive2rel/HiveDbSchema.java:86]: copied.

File: gradle/shipkit.gradle [hunk 6]
```diff
@@ -0,0 +1,39 @@
+//This default Shipkit configuration file was created automatically and is intended to be checked-in.
+//Default configuration is sufficient for local testing and trying out Shipkit.
+//To leverage Shipkit fully, please fix the TODO items, refer to our Getting Started Guide for help:
+//
+//     https://github.com/mockito/shipkit/blob/master/docs/getting-started.md
+//
```
[the remark this answers is not in the document]
Comment (kxu1026) [reply gradle/shipkit.gradle:6]: ok

File: coral-hive/src/main/java/com/linkedin/coral/hive/hive2rel/HiveSchema.java [hunk 7]
```diff
@@ -107,7 +107,7 @@ public Schema snapshot(SchemaVersion schemaVersion) {
   }
 
   /**
-   * Returns Hive metastore client
+   * @return {@link HiveMetastoreClient} object, which represents Hive metastore client
```
[the remark this answers is not in the document]
Comment (kxu1026) [reply coral-hive/src/main/java/com/linkedin/coral/hive/hive2rel/HiveSchema.java:110]: ok

File: gradle/shipkit.gradle [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+shipkit {
+   //TODO is the repository correct?
```
[the remark this answers is not in the document]
Comment (kxu1026) [reply gradle/shipkit.gradle:8]: ok

File: gradle/shipkit.gradle [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+       bintray {
+
+           // The Bintray API token is required to publish artifacts to Bintray
+           // Ensure that the release machine or Travis CI has this env variable exported
+           key = System.getenv("BINTRAY_API_KEY")
+
+           pkg {
+               repo = 'maven'
+               user = 'kxu'
+               userOrg = 'linkedin-coral'
```
[the remark this answers is not in the document]
Comment (kxu1026) [reply gradle/shipkit.gradle:32]: sure. copied.

File: version.properties (hunk 2 above)
Comment (kxu1026) [reply version.properties:4]: Let me verify If I can rollback this. This actually automatically changed by the system.

File: coral-pig/build.gradle [hunk 10]
```diff
@@ -31,16 +30,6 @@ dependencies {
...
-task sourcesJar(type: Jar) {
-  from sourceSets.main.allSource
-  archiveClassifier = 'sources'
-}
-
-task javadocJar(type: Jar) {
-  from javadoc
-  archiveClassifier = 'javadoc'
-}
-
```
[the remark this answers is not in the document]
Comment (kxu1026) [reply coral-pig/build.gradle:43]: same reason which I explained as above. It has been embedded in shipkit plugin.

File: coral-presto/build.gradle [hunk 11]
```diff
@@ -22,36 +22,6 @@ dependencies {
...
-task sourcesJar(type: Jar) {
-  from sourceSets.main.output
-  archiveClassifier = 'sources'
-}
-
-task javadocJar(type: Jar) {
-  from javadoc
-  archiveClassifier = 'javadoc'
-}
-
```
[the remark this answers is not in the document]
Comment (kxu1026) [reply coral-presto/build.gradle:54]: explained. Embedded in shipkit. It must be deleted after shipkit integration.

File: coral-spark/build.gradle [hunk 12]
```diff
@@ -19,16 +19,6 @@ dependencies {
...
-task sourcesJar(type: Jar) {
-  from sourceSets.main.allSource
-  archiveClassifier = 'sources'
-}
-
-task javadocJar(type: Jar) {
-  from javadoc
-  archiveClassifier = 'javadoc'
-}
-
```
[the remark this answers is not in the document]
Comment (kxu1026) [reply coral-spark/build.gradle:31]: because of skipkit intergration.

File: version.properties (hunk 2 above)
Comment (kxu1026) [reply version.properties:4]: I leave this open till next update. I will try and do a local test

File: coral-hive/src/main/java/com/linkedin/coral/hive/hive2rel/HiveDbSchema.java [hunk 13]
```diff
@@ -83,7 +83,7 @@ public RelProtoDataType getType(String s) {
 
   /**
    * A Hive DB does not have subschema
-   * @param name
+   * @param name Subscheme name
```
Comment (wmoustafa) [inline coral-hive/src/main/java/com/linkedin/coral/hive/hive2rel/HiveDbSchema.java:86]: Subshema

File: version.properties [hunk 14]
```diff
@@ -0,0 +1,4 @@
+#Version of the produced binaries. This file is intended to be checked-in.
+#It will be automatically bumped by release automation.
+version=1.0.0
+previousVersion=0.0.1
```
Comment (wmoustafa) [inline version.properties:4]: We discussed offline that we should keep the previous version of this?

Comment (kxu1026) [reply version.properties:4]: fixed

File: coral-hive/src/main/java/com/linkedin/coral/hive/hive2rel/HiveDbSchema.java (hunk 13 above)
Comment (kxu1026) [reply coral-hive/src/main/java/com/linkedin/coral/hive/hive2rel/HiveDbSchema.java]: I am assuming you mean 'Subschema'. :) Fixed.

**Pull request merged**